### PR TITLE
Fix cloud backups

### PR DESCRIPTION
**Pull request by benbromhead**
Fixed cloud backups to support new user-defined config map volumes.
Fixed Helm packages to support backups/user defined config map volumes.
Better error handling for backup API calls.

### CODE REVIEW

File: java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java [hunk 1]
```diff
@@ -128,9 +137,16 @@ private void createOrReplaceBackup(final Backup backup) throws ApiException, Unk
                 .parallelStream()
                 .map(x -> callBackupApi(x, backup))
                 .anyMatch(e -> !e)) {
-            //TODO: Set backup status to PROCESSED
+            BackupSpec backupSpec = backup.getSpec();
+            backupSpec.setStatus("ERROR");
+            backup.setSpec(backupSpec);
+            customObjectsApi.patchNamespacedCustomObject("monitoring.coreos.com", "v1", backup.getMetadata().getNamespace(), "cassandra-backups", backup.getMetadata().getName(), backup);
```
Comment (zegelin) [inline java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java]: `monitoring.coreos.com`?

File: java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java [hunk 2]
```diff
@@ -128,9 +137,16 @@ private void createOrReplaceBackup(final Backup backup) throws ApiException, Unk
...
+        } else {
+            BackupSpec backupSpec = backup.getSpec();
```
Comment (zegelin) [inline java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java]: I think these two if branches can be consolidated. The only difference seems to be the status value -- "ERROR" vs "PROCESSED"?

File: java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java (hunk 1 above)
Comment (benbromhead) [reply java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java]: that would explain the missing status updates

File: java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java (hunk 2 above)
Comment (benbromhead) [reply java/operator/src/main/java/com/instaclustr/cassandra/operator/service/BackupControllerService.java]: Consolidated

**Pull request merged**